boards/common/esp32c3/include/board_common.h

```diff
@@ -0,0 +1,141 @@
+ * ESP32-C3 boards.
+ *
+ * For detailed information about the configuration of ESP32-C3 boards, see
+ * section \ref esp32_peripherals "Common Peripherals".
+ *
+ * @author      Gunar Schorcht <gunar@schorcht.net>
+ * @file
+ * @{
+ */
+
+#ifndef BOARD_COMMON_H
+#define BOARD_COMMON_H
+
+#include <stdint.h>
+
+#include "cpu.h"
+#include "periph_conf.h"
+#if MODULE_ARDUINO
+#include "arduino_pinmap.h"
+#endif
+
+#include "periph/gpio.h"
+#include "sdkconfig.h"
+
+#if MODULE_MTD
+#include "mtd.h"
+#endif
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @name   LED configuration (three predefined LEDs at maximum)
+ *
+ * @note LEDx_ACTIVE value must be declared in board configuration.
+ * @{
+ */
+#if defined(LED0_PIN) || DOXYGEN
+#define LED0_MASK       (BIT(LED0_PIN))
+#define LED0_ON         (gpio_write(LED0_PIN,  LED0_ACTIVE))
+#define LED0_OFF        (gpio_write(LED0_PIN, !LED0_ACTIVE))
+#define LED0_TOGGLE     (gpio_toggle(LED0_PIN))
+#endif
+
+#if defined(LED1_PIN) || DOXYGEN
+#define LED1_MASK       (BIT(LED1_PIN))
+#define LED1_ON         (gpio_write(LED1_PIN,  LED1_ACTIVE))
+#define LED1_OFF        (gpio_write(LED1_PIN, !LED1_ACTIVE))
+#define LED1_TOGGLE     (gpio_toggle(LED1_PIN))
+#endif
+
+#if defined(LED2_PIN) || DOXYGEN
+#define LED2_MASK       (BIT(LED2_PIN))
+#define LED2_ON         (gpio_write(LED2_PIN,  LED2_ACTIVE))
+#define LED2_OFF        (gpio_write(LED2_PIN, !LED2_ACTIVE))
+#define LED2_TOGGLE     (gpio_toggle(LED2_PIN))
+#endif
+/** @} */
+
+/**
+ * @name    STDIO configuration
+ * @{
+ */
+/**< Default baudrate of UART for stdio */
+#ifndef STDIO_UART_BAUDRATE
+#define STDIO_UART_BAUDRATE      (115200)
+#endif
+/** @} */
+
+#if MODULE_MTD || DOXYGEN
+/**
+ * @name    MTD system drive configuration
+ *
+ * Built-in SPI flash memory is used as MTD system drive.
+ * @{
+ */
+
+/**
+ * @brief   MTD drive start address in SPI flash memory
+ *
+ * Defines the start address of the MTD system device in the SPI
+ * flash memory. It can be overridden by \ref esp32_application_specific_configurations
+ * "application-specific board configuration"
+ *
+ * If the MTD start address is not defined or is 0, the first possible
+ * multiple of 0x100000 (1 MByte) is used in free SPI flash memory,
+ * which was determined from the partition table.
+ */
+#ifndef SPI_FLASH_DRIVE_START
+#define SPI_FLASH_DRIVE_START  0
+#endif
+
+/** Default MTD drive definition */
+#define MTD_0 mtd0
+
+/** Pointer to the default MTD drive structure */
+extern mtd_dev_t *mtd0;
+
+/** @} */
+#endif /* MODULE_MTD || DOXYGEN */
+
+#if MODULE_SPIFFS || DOXYGEN
+/**
+ * @name    SPIFFS configuration for the system MTD device
+ *
+ * Configuration of the SPIFFS that can be used on the system MTD device.
+ * @{
+ */
+#define SPIFFS_ALIGNED_OBJECT_INDEX_TABLES 1
+#define SPIFFS_READ_ONLY 0
+#define SPIFFS_SINGLETON 0
+#define SPIFFS_HAL_CALLBACK_EXTRA 1
+#define SPIFFS_CACHE 1
+/** @} */
+#endif /* MODULE_SPIFFS || DOXYGEN */
```

Comment:
Those are the defaults in `pkg/spiffs` anyway, no need to define them again.

Reply:
This is just a copy from here https://github.com/RIOT-OS/RIOT/blob/f375856a7e2d612c0a8f38052fded69d797cd93b/boards/common/esp32/include/board_common.h#L135 which in turn was probably a copy from another board definition when I wrote this common board definition some years ago, maybe from `native`.

So it seems to be time to clean up all ESP32x common board definitions.

BTW, currently the common board definitions for the different ESP32x SoCs contain a lot of duplicate code. They could easily use a single common board definition with slight extensions for the specific ESP32x SoC. This code deduplication is on my ToDo list, but it also requires that the definition of the peripherals has to be done in the same way for all ESP32x boards.

Reply:
I felt free to also commit a change to `boards/common/esp32/include/board_common.h` to this PR.